src/trunk.c

```diff
@@ -6111,9 +6116,8 @@ trunk_compacted_subbundle_lookup(trunk_handle    *spl,
             spl->stats[tid].branch_lookups[height]++;
          }
          if (local_found) {
-            slice message = writable_buffer_to_slice(data);
-            if (data_message_class(spl->cfg.data_cfg, message)
-                != MESSAGE_TYPE_UPDATE) {
+            message msg = merge_accumulator_to_message(data);
+            if (message_class(msg) != MESSAGE_TYPE_UPDATE) {
```

Comment:
We have a bunch of these and the complementary type check elsewhere.

I think it might be both more readable and future-proof if we had a `message_is_final`-type function that make what the purpose of this type of check is explicit.